ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/token/selection/TokenBalanceCriteria.java

```diff
@@ -0,0 +1,193 @@
+
+    /**
+     * The {@link SecureHash} of the issuer of tokens to be selected.
+     */
+    @NotNull
+    private final SecureHash issuerHash;
+
+    /**
+     * The {@link MemberX500Name} of the notary of the tokens to be selected.
+     */
+    @NotNull
+    private final MemberX500Name notaryX500Name;
+
+    /**
+     * The symbol of the notary of tokens to be selected.
+     */
+    @NotNull
+    private final String symbol;
+
+    /**
+     * An optional regular expression to match against the {@link ClaimedToken#getTag()} field, or null to match all tags.
+     */
+    @Nullable
+    private String tagRegex;
+
+    /**
+     * An optional owner {@link SecureHash} of the tokens to be selected, or null to match all owners.
+     */
+    @Nullable
+    private SecureHash ownerHash;
+
+    /**
+     * Creates a new instance of the {@link TokenBalanceCriteria} class.
+     *
+     * @param tokenType      The type of tokens to be selected.
+     * @param issuerHash     The {@link SecureHash} of the issuer of tokens to be selected.
+     * @param notaryX500Name The {@link MemberX500Name} of the notary of the tokens to be selected.
+     * @param symbol         The symbol of the notary of tokens to be selected.
+     */
+    public TokenBalanceCriteria(
+            @NotNull final String tokenType,
+            @NotNull final SecureHash issuerHash,
+            @NotNull final MemberX500Name notaryX500Name,
+            @NotNull final String symbol) {
+        this.tokenType = tokenType;
+        this.issuerHash = issuerHash;
+        this.notaryX500Name = notaryX500Name;
+        this.symbol = symbol;
+    }
+
+    /**
+     * Gets the type of tokens to be selected.
+     *
+     * @return Returns the type of tokens to be selected.
+     */
+    @NotNull
+    public String getTokenType() {
+        return tokenType;
+    }
+
+    /**
+     * Gets the {@link SecureHash} of the issuer of tokens to be selected.
+     *
+     * @return Returns the {@link SecureHash} of the issuer of tokens to be selected.
+     */
+    @NotNull
+    public SecureHash getIssuerHash() {
+        return issuerHash;
+    }
+
+    /**
+     * Gets the {@link MemberX500Name} of the notary of the tokens to be selected.
+     *
+     * @return Returns the {@link MemberX500Name} of the notary of the tokens to be selected.
+     */
+    @NotNull
+    public MemberX500Name getNotaryX500Name() {
+        return notaryX500Name;
+    }
+
+    /**
+     * Gets the symbol of the notary of tokens to be selected.
+     *
+     * @return Returns the symbol of the notary of tokens to be selected.
+     */
+    @NotNull
+    public String getSymbol() {
+        return symbol;
+    }
+
+    /**
+     * Gets an optional regular expression to match against the {@link ClaimedToken#getTag()} field, or null to match all tags.
+     *
+     * @return Returns an optional regular expression to match against the ClaimedToken.tag field, or null to match all tags.
```

Comment:
Is this comment correct? I am not sure to where the link should point to. At the moment, is pointing the ClaimedToken#getTag() but I am not sure if this is correct.